tornadox/blocks/sde/ibm.py

```diff
@@ -28,23 +19,42 @@ def preconditioner_diagonal(*, dt, num_derivatives):
     return scaling_vector, scaling_vector_inv
 
 
+@partial(jax.jit, static_argnames=("num_derivatives",))
+def system_matrices_1d(*, num_derivatives):
+    x = jnp.arange(num_derivatives + 1)
+
+    A_1d = jnp.flip(
+        _arange_to_pascal_matrix(x)[0]
+    )  # no idea why the [0] is necessary...
+    Q_1d = jnp.flip(_arange_to_hilbert_matrix(x))
+    return A_1d, jnp.linalg.cholesky(Q_1d)
+
+
 @jax.jit
-def _hilbert(a):
+def _arange_to_hilbert_matrix(a):
     return 1 / (a[:, None] + a[None, :] + 1)
 
 
 @jax.jit
-def _pascal(a):
-    return _binom(a[:, None], a[None, :])
+def _arange_to_pascal_matrix(a):
+    return _batch_gram(_binom)(a[:, None], a[None, :])
 
 
 def _batch_gram(k):
+    r"""Batch a function.
+
+    A function :math:`f: R^k \times R^k \rightarrow R` becomes a function
+
+        .. math: g: R^{n \times k} \times R^{k \times m} \rightarrow R^{n \times m}.
+
+    The batching follows broadcasting rules:
+    think `jnp.exp(x[:, None], y[None, :])` for 1d inputs.
+    """
     k_vmapped_x = jax.vmap(k, in_axes=(0, None), out_axes=-1)
     k_vmapped_xy = jax.vmap(k_vmapped_x, in_axes=(None, 1), out_axes=-1)
     return jax.jit(k_vmapped_xy)
```

Comment:
How about we rename the input to `f` to match the docstring (also it's a _f_unction)?
```suggestion
def _batch_gram(f):
    r"""Batch a function.

    A function :math:`f: R^k \times R^k \rightarrow R` becomes a function

        .. math: g: R^{n \times k} \times R^{k \times m} \rightarrow R^{n \times m}.

    The batching follows broadcasting rules:
    think `jnp.exp(x[:, None], y[None, :])` for 1d inputs.
    """
    f_vmapped_x = jax.vmap(f, in_axes=(0, None), out_axes=-1)
    f_vmapped_xy = jax.vmap(f_vmapped_x, in_axes=(None, 1), out_axes=-1)
    return jax.jit(f_vmapped_xy)
```